client/src/App.tsx

```diff
@@ -18,14 +20,13 @@ function App(): JSX.Element {
           <AuthProvider>
             <SocketProvider>
               <Switch>
+                <Route exact path="/editProfile" component={EditProfileForm} />
                 <Route exact path="/login" component={Login} />
                 <Route exact path="/signup" component={Signup} />
-                <Route exact path="/dashboard">
+                <Route exact path="/demo" component={Dashboard} />
+                <ProtectedRoute exact path="/dashboard">
                   <Dashboard />
-                </Route>
-                <Route path="*">
-                  <Redirect to="/login" />
-                </Route>
```

Comment:
We should keep this catch-all route around.

Reply:
Noted with thanks